superduper/base/metadata.py

```diff
@@ -240,6 +243,8 @@ class ArtifactRelations(Base):
     'Job': Job,
 }
 
+_component_cache = contextvars.ContextVar('component_cache', default=None)
```

Comment:
Please explain a bit.

Reply:
This make a context for cache, and use None as a flag to check if we are in cahce context.

Difference cache context have difference component_cache